src/mo_frontend/typing.ml

```diff
@@ -2048,10 +2049,24 @@ and detect_lost_fields env t = function
          | Some _ -> ()
          | None ->
             warn env fld.at "M0215"
-              "Field `%s` is ignored in record of type%a"
-              id
+              "Field `%s` is ignored in record%s of type%a"
+              id (if bs = [] then "" else " extension")
               display_typ t)
       flds
+  | ObjBlockE (_exp_opt, { it = Type.Object; _}, _typ_opt, dec_fields) ->
+    let pub_fields = pub_fields dec_fields |> snd in
+    let pub_ids = T.Env.keys pub_fields in
+    let [@warning "-8"] T.Obj (_, fts) = t in
+    List.iter
+      (fun id ->
+        match List.find_opt (fun ft -> ft.T.lab = id) fts with
```

Comment:
```suggestion
        match List.find_opt (fun ft -> ft.T.lab = id) fts with
```
This might confuse typ_fields with value fields. Use Type.find_val_field_opt or what ever it's called.

Reply:
Oh this is in the type, I'll correct and add tests. Maybe we can also warn for ignored public types?

Reply:
Weird, this test doesn't trigger the warning
``` Motoko
let bar_type_no_warn : { foo : Text } = object { public let foo = "FOO"; public type Quux = Text };
```
is it this that you had in mind?

Reply:
Gotcha: this should warn:
``` Motoko
let bar_type_warn : { type Quux = Text; foo : Text} = object { public let foo = "FOO"; public let Quux = "QUUX"; public type Quux = Text };
```
but doesn't a.t.m.

Reply:
0e00886b5 fixes this.